docs/card-rendering.md

```diff
@@ -0,0 +1,19 @@
+# Card Rendering
+
+Since cards are neither HTML nor Ember components, they cannot be directly compiled by browsers or Glimmer. Therefore, to compile a card, it needs to be transformed from a card instance into an Ember component and then served as HTML. This transformation process primarily occurs in the [`packages/host/app/services/render-service.gts`](../packages/host/app/components/card-prerender.gts)which utilizes an [isolated `render()` function](../packages/host/app/lib/isolated-render.gts). This render() function offers a Glimmer rendering context, which keeps its errors isolated from the main Glimmer render process that renders the host app.
+
```

Comment:
this is not really correct. it is true that the gts files are not native to browser, but we deal with that by transpiling the gts files into normal ES6 modules in the realm server (here https://github.com/cardstack/boxel-motion/blob/main/packages/runtime-common/realm.ts#L434-L484), and specifically the js that is tranpiled does actually include ember components (actually more precisely glimmer components) used to render the card into the page. The HTML that we generate as part of prerendering will be used as an optimization when displaying hetrogenous card results, the idea is that when we have a page will with lots of different cards, we can just render the HTML instead of having to load the code for each card. We currently aren't using the prerendered card HTML, but we intend to shortly. The other more important job of this service is to actually drive the indexing such that the load for field's value that we try to render into the pre-rendered HTML actually is reflected in a "search doc" that lives in the indexer which facilitates searching for cards.